plugin/src/main/java/org/openrewrite/gradle/RewritePlugin.java

```diff
@@ -98,10 +100,16 @@ private static void configureProject(Project project, RewriteExtension extension
             }
 
             //Collect Java metadata for each project (used for Java Provenance)
-            //Using the older javaConvention because we need to support older versions of gradle.
-            @SuppressWarnings("deprecation")
-            JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
-            javaConvention.getSourceSets().all(sourceSet -> {
+            SourceSetContainer sourceSets;
+            if (project.getGradle().getGradleVersion().compareTo("7.0") > 0) {
+                sourceSets = project.getExtensions().getByType(JavaPluginExtension.class).getSourceSets();
+            } else {
+                //Using the older javaConvention because we need to support older versions of gradle.
+                @SuppressWarnings("deprecation")
+                JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
+                sourceSets = javaConvention.getSourceSets();
```

Comment:
This will work for now, but if 9.0 actually removes the conventions or `JavaPluginConvention`, then this code will no longer compile when the Gradle wrapper is upgraded to 9.0 in this repo and will require being refactored again.

Reply:
> this code will no longer compile when the Gradle wrapper is upgraded to 9.0 in this repo and will require being refactored again.

Yes, but I think released versions will still works fine with 9.0+, and no warning for 7.1+.